src/libAtomVM/opcodesswitch.h

```diff
@@ -420,6 +420,10 @@ term make_fun(Context *ctx, const Module *mod, int fun_index)
     uint32_t n_freeze = module_get_fun_freeze(mod, fun_index);
 
     int size = 2 + n_freeze;
+    if (memory_ensure_free(ctx, size + 1) != MEMORY_GC_OK) {
+        fprintf(stderr, "Unable to ensure %i free terms %s:%i\n", size + 1, __FILE__, __LINE__);
+        abort();
```

Comment:
we should raise an error here instead of aborting the whole virtual machine.

Reply:
RAISE_ERROR is a macro, which can't be used here.  So I changed the code to return the nil term if there is a memory error, and check for that value at the only call point of the function, and use the RAISE_ERROR macro in the case statement, where the macro can be used.